Lib/collections/__init__.py

```diff
@@ -727,11 +727,8 @@ def __eq__(self, other):
             return NotImplemented
         return all(self[e] == other[e] for c in (self, other) for e in c)
 
-    def __ne__(self, other):
-        'True if any counts disagree. Missing counts are treated as zero.'
-        if not isinstance(other, Counter):
-            return NotImplemented
-        return not self == other
+    # Restore the default implementation, override dict.__ne__
+    __ne__ = object.__ne__
```

Comment:
Jumping mro from Counter to past dict directly to object seem like a bit of hack, one that we might regret for subclasses using *super()*  Also, the current docstring is useful.   If you really have to change *something*, we could just delete lines 733 and 734, leaving the instance() check for \__eq__().  But that does mess-up the parallelism a bit and it looks odd so I would rather not do that either.

Reply:
If you want to keep the docstring and make `__ne__` subclass friendly, you can reimplement the logic of `object.__ne__`:
```python
    def __ne__(self, other):
        'True if any counts disagree. Missing counts are treated as zero.'
        eq = self.__eq__(other)
        return NotImplemented if eq is NotImplemented else not eq
```
or better reuse `object.__ne__`:
```python
    def __ne__(self, other):
        'True if any counts disagree. Missing counts are treated as zero.'
        return object.__ne__(self, other)
```
But would not be better to improve the `object.__ne__` docstring so that it will be good for Counter and all other classes with customized equality comparison? BTW the current docstring of `object.__ne__` is not correct because the special method can return NotImplemented.